Interface.C

```diff
@@ -268,41 +327,39 @@ void preciceAdapter::Interface::writeCouplingData()
 {
     // TODO: wrap around isWriteDataRequired
     // Does the participant need to write data or is it subcycling?
-    // if (precice_.isWriteDataRequired(computedTimestepLength))
-    // {
-        // Make every coupling data writer write
-        for (uint i = 0; i < couplingDataWriters_.size(); i++)
-        {
-            // Pointer to the current reader
-            preciceAdapter::CouplingDataUser *
+    // Make every coupling data writer write
+    for (uint i = 0; i < couplingDataWriters_.size(); i++)
+    {
+        // Pointer to the current reader
+        preciceAdapter::CouplingDataUser *
                 couplingDataWriter = couplingDataWriters_.at(i);
 
-            // Write the data into the adapter's buffer
-            couplingDataWriter->write(dataBuffer_);
+        // Write the data into the adapter's buffer
+        couplingDataWriter->write(dataBuffer_, provideMeshConnectivity_);
 
-            // Make preCICE write vector or scalar data
-            if (couplingDataWriter->hasVectorData())
-            {
-                precice_.writeBlockVectorData
-                (
-                    couplingDataWriter->dataID(),
-                    numDataLocations_,
-                    vertexIDs_,
-                    dataBuffer_
-                );
-            }
-            else
-            {
-                precice_.writeBlockScalarData
-                (
-                    couplingDataWriter->dataID(),
-                    numDataLocations_,
-                    vertexIDs_,
-                    dataBuffer_
-                );
-            }
+        // Make preCICE write vector or scalar data
+        if (couplingDataWriter->hasVectorData())
+        {
+            precice_.writeBlockVectorData
+                    (
+                        couplingDataWriter->dataID(),
+                        numDataLocations_,
+                        centerIDs_,
```

Comment:
This probably affects how we currently can read/write to both faceCenters and faceNodes.

I tried the FSI flap_perp tutorial and it runs, but I am not confident that it does not introduce any error or problems. Please look at this closer.

Reply:
Actually, this is not a problem, because it is only `vertexIDs` renamed to `centerIDs`. I would rename them back to `vertexIDs`, to use the preCICE terminology.

Reply:
Correct, relic from the old approach. Fixed.